### ISSUE TITLE
Extend Referrence Documentation for bivariate

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/hvplot#835]
I'm trying to add docstrings to the `core` module. This include `.bivariate`. 

In general I believe the Reference documentation is way too sparse and does not help users understand the what and how of the specific plot. But [the reference documentation for `bivariate`](https://hvplot.holoviz.org/reference/pandas/bivariate.html#pandas-gallery-bivariate) I've spend a considerable long time just understanding what type of plot this is. And I am still not sure.

![image](https://user-images.githubusercontent.com/42288570/185293568-ee1cad99-64da-4a3f-87b3-f5dc2f7acf5e.png)

- When I [google search](https://www.google.com/search?q=bivariate+plot) for bivariate plot most of the references show something that looks close to scatter plots.
- Plotly calls this a [2D Histogram Contour](https://plotly.com/python/2d-histogram-contour/) I believe?. It is very hard for me to be sure as the hvPlot Reference does not describe the method or provide a link to a reference explanation. If I search wikipedia I again find something that does not look like a contour plot https://en.wikipedia.org/wiki/Bivariate_analysis

As 

- the reference documentation is very sparse
- the name `bivariate` and short description is very unclear
- the underlying code is very difficult to follow
- Pandas `.plot` does not seem to have  a similar `kind` of plot

It is so hard to understand what this is and how to use it.

My guess is that it is possible for example to add a `reduce_func` to specify the kind of aggregation done? At least plotly supports that. But I cannot find it via `hvplot.help('bivariate')`. But `group` and `groupby` are not documented there either. So I don't trust that help information 100%.

I could add similar comments for any of the reference documentation and docstrings. Its like the documentation and implementation details are great for developers but has never been reviewed by a user. For a user it is so far from great.

**Comment by MarcSkovMadsen** [comment]
The best reference I've been able to stumble upon is this one I believe https://bio304-class.github.io/bio304-fall2017/ggplot-bivariate.html. But it also spends a considerable amount of time educating its users on bivariate plots.

**Comment by droumis** [comment]
I agree.. it seems like it should be called 'bivariate density' or '2D density contour' plot instead, but these are so wordy. I think improving the documentation for it is the correct first step here
